Plugins/Geant4/src/Geant4DetectorSurfaceFactory.cpp

```diff
@@ -26,12 +26,13 @@ void Acts::Geant4DetectorSurfaceFactory::construct(
   auto g4Translation = g4PhysVol.GetTranslation();
   auto g4Rotation = g4PhysVol.GetRotation();
 
-  G4Transform3D g4Transform =
-      (g4Rotation == nullptr)
-          ? G4Transform3D(CLHEP::HepRotation(), g4Translation)
-          : G4Transform3D(*g4Rotation, g4Translation);
+  if (g4Rotation == nullptr) {
+    g4Rotation = new CLHEP::HepRotation();
```

Comment:
`new` without `delete`?

Reply:
Missed this one. Explicit `delete` won't cut it here, so I changed it to something resembling the previous solution.